### ISSUE TITLE
Views UI: Edit Basic Settings

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2906]
## Describe your issue or idea

The title of this issue is basically a module: https://www.drupal.org/project/views_ui_basic

> Does your client want to modify the header, footer, title, or empty text of a view, but you don't want to train them on the rather intimidating Views UI admin interface or give them access to pages that may allow them to mess up their site? Then this handy little module is the solution for you!

> "Views UI: Edit Basic Settings" places edit tabs on Views pages, similar to node pages, and allows users with the correct permission to modify their header, footer, title, empty text, or number of items to display. This module also provides a separate interface that displays a list of views. All Views are defined by you, so you can exclude certain Views.

I believe that adding this functionality in Backdrop core would solve a pain point that many site builders face when their customers request access to edit views-provided content. We don't want to allow them to shoot themselves on the foot, but there are basic settings that they should have the ability to edit at will.

<img width="729" alt="screen shot 2017-11-13 at 7 10 47 am" src="https://user-images.githubusercontent.com/2423362/32702897-cfc8dbf2-c841-11e7-9823-186546185c2d.png">

<img width="731" alt="screen shot 2017-11-13 at 7 17 01 am" src="https://user-images.githubusercontent.com/2423362/32702952-c4f6c184-c842-11e7-9e1d-bed03aeed216.png">

<img width="788" alt="screen shot 2017-11-13 at 7 14 36 am" src="https://user-images.githubusercontent.com/2423362/32702937-5d058394-c842-11e7-8e33-7cf123a13502.png">

<img width="1213" alt="screen shot 2017-11-13 at 7 15 31 am" src="https://user-images.githubusercontent.com/2423362/32702942-7a24f446-c842-11e7-9400-2b190906ec41.png">

**Comment by olafgrabienski** [comment]
I'd find such a permission very helpful. One example: At the moment I don't know a reasonable approach how to let an editor change content above a view without assigning too extensive permissions. Views or Block / Layout related permissions don't fit really for such a use case.

That said, the suggested approach seems to allow more than needed. See for instance the "Replacement patterns" tab in the last screenshot above. That's distractiing for an editor (and gives the option to shoot her /himself in the foot).  So, if we add the functionality I'd suggest to streamline the UI to adapt it even more for an editorial use case.

**Comment by laryn** [comment]
I agree, this would be great!

**Comment by klonos** [comment]
I think that one can achieve such a thing using paragraphs, but we should consider providing this in core as a "lightweight" alternative.

**Comment by olafgrabienski** [comment]
@klonos I agree, a lightweight and easy to find solution would be nice. I guess however that the editorial interface will look quite cluttered with the suggested functionality. What do you think about my proposal above to adapt the UI for an editorial use case?

**Comment by klonos** [comment]
@olafgrabienski sorry for the belated reply...

Yes, I *personally* think that content editors should only be limited to the actual content of the header. Having said that though, I understand that the use cases might be different for others. I believe that the granularity of configuration the views_ui_basic module offers to site admins is perfect. You can limit the content editors to only be able to edit the actual header/footer content, thus preventing changes to the admin label of the header/footer area, the "display even if empty" setting, empty text, pager etc. If your content editors are either more tech-savvy, or they require access to these settings, then you can do that too.

I agree that replacement tokens can be dangerous to expose to content editors, as well as confusing. The site admin UX problems I see if we merged that module in core as is, are the following:

1. the module allows you to select which display of which view will have editable components, but which components are editable is not configurable per view; it applies to all views/display that are selected

2. one can have multiple headers/footers in each display of each view. The views_ui_basic module automatically grants edit permission to all headers or all footers of all these header/footer areas.

I see two possible solutions:

A) we build something along the lines of the views_ui_basic module, but we make sure that the editable header/footer thing is per header/footer item added, per display.

B) we make it so that specific content types can be added/attached/referenced to be used as footer/header. Content editors would then simply need to edit these same as they do for other content.

Solution B above resembles adding custom content blocks before/after content so that they can be used as the view header/footer, but then do we have a good editorial UX for editing these blocks?

Related: #1673

https://www.drupal.org/project/views/issues/1708320

> When creating views as pages, one has the option to add a text area with any html markup, to appear on top or below the view (known as "header" and "footer").
> 
> However, when the user has installed external WYSIWYG editors to facilitate this html markup, it seems that Views does not allow WYSIWYG editors here [on purpose](https://www.drupal.org/node/1698224#comment-6267276).
>...
>Enabling this leads to various bugs and major shitstorms: http://drupal.org/node/1065344

https://www.drupal.org/node/1698224#comment-6267276

>In Views for Drupal 7 the textarea's `#wysiwyg` attribute is set to `FALSE` for header, footer and other textareas. This means that Views developers don't want to have any WYSIWYG editor there. We should respect this and this is why you don't see CKEditor there. ...
